Signal/src/ViewControllers/NewContactThreadViewController.m

```diff
@@ -113,9 +113,25 @@ - (void)loadView
     [self.noSignalContactsView autoPinEdgeToSuperviewEdge:ALEdgeTop];
     [self.noSignalContactsView autoPinToBottomLayoutGuideOfViewController:self withInset:0];
 
+    UIRefreshControl *pullToRefreshView = [UIRefreshControl new];
+    pullToRefreshView.tintColor = [UIColor grayColor];
+    [pullToRefreshView addTarget:self
+                          action:@selector(pullToRefreshPerformed:)
+                forControlEvents:UIControlEventValueChanged];
+    [self.tableViewController.tableView insertSubview:pullToRefreshView atIndex:0];
+
     [self updateTableContents];
 }
 
+- (void)pullToRefreshPerformed:(UIRefreshControl *)refreshControl
+{
+    OWSAssert([NSThread isMainThread]);
+
+    [self.contactsViewHelper.contactsManager requestSystemContactsOnce];
```

Comment:
As I understand it, the primary use case for this feature is when Alice has had Bob in her contacts for a long time, but Bob *just* signed up for Signal. Alice won't see Bob listed as a valid recipient (with HEAD of master) unless:
1.  she knows to restart her app
2. she makes some edit to her contacts
3. the contact-refresh timeout expires (12 hours).

As I understand it the purpose of this feature is to give a user a somewhat intuitive way to override that 12 hour timeout. As written, I don't think this change would actually accomplish anything since we already `requestSystemContactsOnce` when this VC appears.

Instead I think we want this to perform a contact intersection **every** time someone pulls to refresh, i.e. call `updateContacts` directly. Though maybe we need to ensure we're doing the right thing WRT contact access.

Reply:
Great point.  I've updated the PR.  

PTAL @michaelkirk